server/src/service/collection.rs

```diff
@@ -0,0 +1,201 @@
+        }
+
+        let _collection = KEYRING
+            .get()
+            .unwrap()
+            .create_item(label, attributes.clone(), secret, replace)
+            .await;
+
+        // lookup just created item
+        let lookup_item = KEYRING
+            .get()
+            .unwrap()
+            .lookup_item(&attributes)
+            .await
+            .unwrap()
+            .unwrap();
+        // make prompt to get the secret and set it with set_secret()
+
+        let created_item_path = ObjectPath::default().into(); // how to get path from _collection
+        let prompt = ObjectPath::default().into(); // temp Prompt
+
+        let mut attributes_for_new: HashMap<&str, &str> = Default::default();
+        for (key, value) in lookup_item.attributes() {
+            attributes_for_new.insert(key.as_str(), value);
+        }
+
+        // TODO map portal::Item to crate::Item and push it to items attribute
+        let item = item::Item::new(
+            &attributes_for_new,
+            lookup_item.secret().to_vec(),
+            lookup_item.label(),
+            lookup_item.created().as_secs(),
+            lookup_item.modified().as_secs(),
+            self.path().await,
+        )
+        .await;
+        self.items.push(item);
+
+        (created_item_path, prompt)
+    }
+
+    #[dbus_interface(property, name = "Items")]
+    async fn items(&self) -> Vec<ObjectPath> {
+        let mut paths: Vec<ObjectPath> = Vec::new();
+
+        for path in &self.items {
+            paths.push(path.path().await.into());
+        }
+
+        paths
+    }
+
+    #[dbus_interface(property, name = "Label")]
+    pub async fn label(&self) -> &String {
+        &self.label
+    }
+
+    #[dbus_interface(property, name = "Locked")]
+    pub async fn locked(&self) -> bool {
+        self.locked
+    }
+
+    #[dbus_interface(property, name = "Created")]
+    pub async fn created(&self) -> u64 {
+        self.created
+    }
+
+    #[dbus_interface(property, name = "Modified")]
+    pub async fn modified(&self) -> u64 {
+        self.modified
+    }
+
+    #[dbus_interface(signal)]
+    pub async fn item_created(ctxt: &SignalContext<'_>) -> Result<(), Error>;
+
+    #[dbus_interface(signal)]
+    pub async fn item_deleted(ctxt: &SignalContext<'_>) -> Result<(), Error>;
+
+    #[dbus_interface(signal)]
+    pub async fn item_changed(ctxt: &SignalContext<'_>) -> Result<(), Error>;
+}
+
+impl Serialize for Collection {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        OwnedObjectPath::serialize(&self.path, serializer)
+    }
+}
+
+impl Type for Collection {
+    fn signature() -> Signature<'static> {
+        ObjectPath::signature()
+    }
+}
+
+impl Collection {
+    // temporarily creates a generic Collection object
+    pub async fn new(label: String) -> Self {
+        Collection {
+            items: Vec::new(),
+            label: label.clone(),
```

Comment:
```suggestion
            label: label.to_owned(),
```
Use ToOwned trait when you are taking ownership of the passed data instead of clone.

Reply:
understood, thanks!